PowerEditor/src/ScintillaComponent/AutoCompletion.cpp

```diff
@@ -233,7 +226,7 @@ void AutoCompletion::getWordArray(vector<generic_string> & wordArray, TCHAR *beg
 
 	size_t docLength = _pEditView->execute(SCI_GETLENGTH);
 
-	int flags = SCFIND_WORDSTART | SCFIND_MATCHCASE | SCFIND_REGEXP | SCFIND_POSIX;
+	int flags = SCFIND_WORDSTART | (_ignoreCase ? 0 : SCFIND_MATCHCASE) | SCFIND_REGEXP | SCFIND_POSIX;

```

Comment:
`getWordArray` method is for getting word list for word completion, but has nothing to do with function list of auto-completion.

Reply:
`getWordArray` is used in `AutoCompletion::showApiAndWordComplete()` and I noticed it was fixed  with `SCFIND_MATCHCASE` so that is what I changed 1st. Then I found still issues with the `for` loop with the autocomplete list so I found that the `break` ing out of the loop too early appears to be the issue. If handling `SCFIND_MATCHCASE` setting is not desired, then that could be undone. I considered handling `SCFIND_MATCHCASE` as OK, but perhaps consider as not needed now.

Reply:
I just tested with restoring

```
int flags = SCFIND_WORDSTART | SCFIND_MATCHCASE | SCFIND_REGEXP | SCFIND_POSIX;
```

and the output in DebugView is the same, so this line is not the main issue for the autocomplete so it can remain as it was before.

Reply:
@mpheath 
My bad. It was set on `Function Completion` instead of `Function and Word Completion`. Sorry for the wrong accusation.
It seems "showApiAndWordComplete" that you modified is more accurate than "showApiComplete" now.
However, I have to test it more - to prevent from regression

On the meanwhile, please revert `(_ignoreCase ? 0 : SCFIND_MATCHCASE)` change.
Word completion by default is case sensitive, and it has nothing to do with the function completion's setting.

Reply:
Revert done. We both made some mistakes. Appreciate the concern.